components/iostreams/src/server/output_stream.cpp

```diff
@@ -81,7 +81,7 @@ namespace hpx { namespace iostreams { namespace server
         pending_output_.output(locality_id, count, in, write_f, mtx_);
 
         // Wake up caller.
-        threads::set_thread_state(caller, threads::pending);
+        threads::set_thread_state(caller, threads::thread_state_enum::pending);
```

Comment:
Would it be appropriate to use this opportunity to rename the enum to `hpx::threads::thread_state`?

Reply:
That's what I wanted to do first :P we already have a `thread_state` though, which is the combination of `thread_state_enum` and `thread_state_ex_enum` (I was very confused why they'd need `enum` in the name until I remembered that): https://github.com/STEllAR-GROUP/hpx/blob/e907e3227dd0b22e4e3c7c662a608490de8adb5b/libs/core/coroutines/include/hpx/coroutines/thread_enums.hpp#L139-L141.

I think the combined state is probably the rightful owner of the `thread_state` name (not very strongly, I'm open to alternatives here). That said, I realize now that `thread_state_ex_enum` definitely doesn't need the `enum` except for symmetry with the other enum. The main enum could be `thread_schedule_state`? And the extended one `thread_wait_state`? Or `thread_restart_state`? Or something else?

In any case, a rename would indeed be appropriate.

Reply:
> That's what I wanted to do first :P we already have a `thread_state` though, which is the combination of `thread_state_enum` and `thread_state_ex_enum` (I was very confused why they'd need `enum` in the name until I remembered that):
> 
> https://github.com/STEllAR-GROUP/hpx/blob/e907e3227dd0b22e4e3c7c662a608490de8adb5b/libs/core/coroutines/include/hpx/coroutines/thread_enums.hpp#L139-L141
> 
> .
> I think the combined state is probably the rightful owner of the `thread_state` name (not very strongly, I'm open to alternatives here). That said, I realize now that `thread_state_ex_enum` definitely doesn't need the `enum` except for symmetry with the other enum. The main enum could be `thread_schedule_state`? And the extended one `thread_wait_state`? Or `thread_restart_state`? Or something else?
> 
> In any case, a rename would indeed be appropriate.

Fair points. I withdraw my suggestion. Yours are acceptable.

Reply:
I updated these to be `thread_schedule_state` and `thread_wait_state` now.